builder/builder-source-bzr.c

```diff
@@ -239,6 +252,62 @@ builder_source_bzr_extract (BuilderSource  *source,
   return TRUE;
 }
 
+static gboolean
+builder_source_bzr_bundle (BuilderSource  *source,
+                           BuilderContext *context,
+                           GError        **error)
+{
+  BuilderSourceBzr *self = BUILDER_SOURCE_BZR (source);
+
+  g_autoptr(GFile) sources_dir = NULL;
+  g_autoptr(GFile) dest_dir = NULL;
+  g_autoptr(GFile) dest_dir_tmp = NULL;
+  g_autoptr(GFile) bzr_sources_dir = NULL;
+
+  g_autofree char *sources_dir_path = NULL;
+  g_autofree char *dest_dir_path = NULL;
+  g_autofree char *dest_dir_path_tmp = NULL;
+  g_autofree char *bzr_sources_dir_path = NULL;
+
+  g_autofree char *base_name = NULL;
+  g_autofree char *base_name_tmp = NULL;
+  g_autofree char *app_dir_path = g_file_get_path (builder_context_get_app_dir (context));
```

Comment:
Nitpick: you can initialize `app_dir_path` and `sources_dir_path` right before `bzr_sources_dir_path` instead